crates/libtiny_client/src/state.rs

```diff
@@ -643,15 +651,15 @@ impl StateInner {
             AUTHENTICATE { ref param } => {
                 if param.as_str() == "+" {
                     // Empty AUTHENTICATE response; server accepted the specified SASL mechanism
-                    // (PLAIN)
                     if let Some(ref auth) = self.server_info.sasl_auth {
-                        let msg = format!(
-                            "{}\x00{}\x00{}",
-                            auth.username, auth.username, auth.password
-                        );
-                        snd_irc_msg
-                            .try_send(wire::authenticate(&base64::encode(&msg)))
-                            .unwrap();
+                        let msg = match auth {
+                            SASLAuth::Plain { username, password } => {
+                                let msg = format!("{}\x00{}\x00{}", username, username, password);
+                                base64::encode(&msg)
+                            }
+                            SASLAuth::External { .. } => "+".to_string(),
```

Comment:
Where is this part documented? It would be good to link here to a documentation, I found it a bit hard to find.

Reply:
https://github.com/irssi/irssi/blob/master/src/irc/core/sasl.c#L205

..."the code is self-documenting" 😵‍💫

Reply:
Actually, I found it here: https://ircv3.net/specs/extensions/sasl-3.1.html

> Empty (zero-length) responses are sent as AUTHENTICATE +.

Reply:
Added in latest commit.